### PR TITLE
Migrate react preset and create-project to properly use RHL v4

### PR DESCRIPTION
**Pull request by eliperelman**

### CODE REVIEW

**Comment by edmorley** [comment]
(For reviewing reference) Migration guide:
https://github.com/gaearon/react-hot-loader#migrating-from-v3

File: packages/react/index.js [hunk 1]
```diff
@@ -10,16 +10,21 @@ module.exports = (neutrino, opts = {}) => {
   const mode = neutrino.config.get('mode');
   const options = merge({
     hot: true,
-    hotEntries: [require.resolve('react-hot-loader/patch')],
```
Comment (edmorley) [inline packages/react/index.js]: There are now no default `hotEntries` in any of the presets that inherit `web` (or in `web` itself). Should we now remove `hotEntries` from the web preset entirely?

File: packages/react/README.md [hunk 2]
```diff
@@ -308,30 +310,20 @@ First, install `react-hot-loader` as a dependency, this **must** be React Hot Lo
...
-from `react-hot-loader`. The main file may be named `index.js` or `index.jsx`. The extension is resolved by webpack.
-- Wrap your top-level React component in the `AppContainer`.
-- Perform the application render in a reusable function for initial load and subsequent reloads.
-- Add the `hot` acceptance to call this function.
+Neutrino will then automatically load React Hot Loader then next time a build
+is performed. Next, you need to mark your root component as _hot-exported_.
 
-For example:
+For example, if your `src/index` main entry renders a root component from
+`App.jsx`, then the exported component needs to have a hot export:
```
Comment (edmorley) [inline packages/react/README.md]: Should we add a link to the RHL docs?

**Review by edmorley — APPROVED** [review]
Yey for more things updated! 😃

File: packages/react/README.md (hunk 2 above)
Comment (eliperelman) [reply packages/react/README.md]: Yeah, that would be good.

File: packages/react/index.js (hunk 1 above)
Comment (eliperelman) [reply packages/react/index.js]: I'm thinking so.

File: packages/web/index.js [hunk 3]
```diff
@@ -219,17 +218,6 @@ module.exports = (neutrino, opts = {}) => {
       config.devtool('cheap-module-eval-source-map');
       config.when(options.hot, () => {
         neutrino.use(hot);
-        config.when(options.hotEntries, (config) => {
```
Comment (edmorley) [inline packages/web/index.js]: I think we should add a check for `hotEntries` so the user knows to clean up their config (and also why things aren't working if they rely on it) - like we do for the deprecated `options.minify` options above.

Comment (eliperelman) [reply packages/web/index.js]: Good call.

**Comment by eliperelman** [comment]
Updated PR to upgrade create-project's install of RHL to v4.

**Pull request merged**